File: pset6/readability.py
from cs50 import get_string
s=get_string("Text: ")
sentence=0
word=1
letter=0
for i in s:
    if i==' ':
        word+=1
    elif i.isalpha():
        letter+=1;
    elif i=='?' or i=='.' or i=='!':
        sentence+=1;
index=round(0.0588 * letter/(word/100)- 0.296 /(word/100)* sentence - 15.8)
if index<1:
    print("Before Grade 1")
elif index>=1 and index<=15:
    print(f"Grade {index}")
elif index>=16:
    print("Grade 16+")

File: pset2/readability.c
#include <cs50.h>
#include <stdio.h>
#include <math.h>
#include <ctype.h>
#include <string.h>
int main(void)
{
    string s=get_string("Text: ");
    int n=strlen(s);
    /* f there is 70 space, there is 71 words So space is initialized as 1 */
    int space=1,letters=0,sentences=0;
    for (int i = 0; i<n;i++)
    {
        if (isblank(s[i]))
        {
            space+=1;
        }
        else if (isalnum(s[i]))
        {
            letters+=1;
        }
        else if (s[i]=='?' || s[i]=='.' || s[i]=='!')
        {
            sentences+=1;
        }
    }
    /* round off the Coleman-Liau index*/
    int index=round(0.0588 * (100*letters)/space - 0.296 * (100*sentences)/space - 15.8);
    if (index<=0)
    {
        printf("Before Grade 1\n");
    }
    else if (index>=1 && index <=15)
    {
        printf("Grade %i\n",index);
    }
    else
    {
        printf("Grade 16+\n");
    }
}